### PR TITLE
Fixed two NullPointerException

### PR DESCRIPTION
**Pull request by BrainStone**
Caused by insufficient initialization of the respective objects

### CODE REVIEW

File: server/src/main/java/de/jackwhite20/cobra/server/http/Request.java [hunk 1]
```diff
@@ -71,6 +71,9 @@ private void parse() {
                 headers.put(keyVal[0].trim(), keyVal[1].trim());
             }
         }
+        
+        body(raw.bytes());
```
Comment (JackWhite20) [inline server/src/main/java/de/jackwhite20/cobra/server/http/Request.java]: The body here is actually not the String 'raw'. 'raw' is the HTTP request header with method, path,version and the headers from the request.

Do you get the NPE if body is null there when getting a post form key with 'postForm(..)'?

**Comment by BrainStone** [comment]
Both body and postForm are null after intialization. And since body is null postForm cannot be instatiated.

So I suggest you instantiate both right during construction of the object.

**Comment by BrainStone** [comment]
I’ll make another commit that’ll fix using the wrong variable.

**Comment by JackWhite20** [comment]
I will close this pull request and have a look at the logic to fix the root cause of it. Then you don't have to make another pull request.

**Comment by JackWhite20** [comment]
Are you trying to use a request body with a GET request or something? The body is currently only set when a POST request comes in.

**Comment by BrainStone** [comment]
Well the core issue is that both `body` and `postForm` can be null. In the current implementation this throws a NPE when calling `public String postForm(String key)` (https://github.com/JackWhite20/Cobra/blob/master/server/src/main/java/de/jackwhite20/cobra/server/http/Request.java#L116) and `public Map<String, String> postForm()` (https://github.com/JackWhite20/Cobra/blob/master/server/src/main/java/de/jackwhite20/cobra/server/http/Request.java#L128 (Yes `Collections.unmodifiableMap(null)` throws a NPE))

So I suggest that you make sure that both objects do get a default initialization. At least the `formData` should always be an empty map.

**Comment by JackWhite20** [comment]
I know when and why it is happening, i meant that this only happens when you try to access body data from a NONE POST request. Current logic is to lazy load the post form data when really accessing/using it. When trying to access this post form data on a none POST request, it is fault usage, i could make it throwing an exception or something like that.

**Comment by BrainStone** [comment]
Just initialize the objects with empty default values. That would make a much more smooth user experience.

**Comment by JackWhite20** [comment]
Using the body on a none POST request isn't a good practice anyway, so a proper exception would help instead of just empty values which will not tell the user what is "wrong". I will look what i will change.

**Comment by BrainStone** [comment]
You don't need to throw an exception. Just returning null would probably be better.

**Comment by BrainStone** [comment]
Anyways, a proper exception would be an illegal state exception.

**Comment by BrainStone** [comment]
Any updates on this issue?

**Comment by JackWhite20** [comment]
Yes, I will push a change in the evening.

**Comment by BrainStone** [comment]
Great. Will you push a release or just a snapshot?

**Comment by JackWhite20** [comment]
I will look above the complete source and if I make other changes i will probably make a new release then.

**Comment by BrainStone** [comment]
Alright. Thank you!

**Comment by JackWhite20** [comment]
Is that what you meant?

https://github.com/JackWhite20/Cobra/commit/17c50ed5c1e4c76b0f34372da71335d94cb47413

**Comment by BrainStone** [comment]
Excatly!

**Comment by BrainStone** [comment]
Neither repo contains the commit so far. Would you mind uploading to either?

**Comment by JackWhite20** [comment]
Hey, sorry, it should now be available as 2.3.3-RELEASE.

**Pull request closed without merging**